java-frontend/src/main/java/org/sonar/java/xml/maven/PomParser.java

```diff
@@ -68,6 +69,22 @@ public static MavenProject parseXML(File file) {
     return null;
   }
 
+  /**
+   * By default, the location of XML element is enabled, except on IBM JVM where it is disabled and has to be manually enabled.
+   * The property is a IBM-specific property, not recognized by non-IBM JVMs, and will throw an {@link IllegalArgumentException}.
+   *
+   * See {@link https://www.ibm.com/support/knowledgecenter/SSYKE2_8.0.0/com.ibm.java.zos.80.doc/user/xml/xlxpj_reference.html}.
+   *
+   * @param factory
+   */
+  private static void enableLocationPropertyForIBM(XMLInputFactory factory) {
+    try {
```

Comment:
You should be able to use `isPropertySupported` API on `XMLInputFactory` ,  that way you can perhaps avoid try-catch statement. But I am not sure if it works with IBM JDK, I just saw it in the javadoc.